### PR TITLE
Feature that let the city and the country names be saved in lower case

### PR DESCRIPTION
**Pull request by ghilyana**

### CODE REVIEW

File: src/main/java/com/ghilly/web/handler/CountryHandler.java [hunk 1]
```diff
@@ -60,9 +60,11 @@ public CityDAO getCapitalByCountryId(int countryId) {
         return countryServiceRest.getCapitalByCountryId(countryId);
     }
 
-    private void checkNameIsValid(String name) {
-        checkNameIsWrong(name);
-        checkNameExists(name);
+    private void checkNameIsValid(CountryDAO countryDAO) {
```
Comment (gregorusanov) [inline src/main/java/com/ghilly/web/handler/CountryHandler.java]: please bring only name for country

File: src/main/java/com/ghilly/web/handler/CountryHandler.java [hunk 2]
```diff
@@ -60,9 +60,11 @@ public CityDAO getCapitalByCountryId(int countryId) {
...
+        checkNameIsWrong(countryDAO.getName());
+        String nameInLowerCase = countryDAO.getName().toLowerCase();
```
Comment (gregorusanov) [inline src/main/java/com/ghilly/web/handler/CountryHandler.java]: Set name in lower register before checking

File: src/test/java/com/ghilly/web/handler/CityHandlerTest.java [hunk 3]
```diff
@@ -15,11 +15,13 @@
 
 class CityHandlerTest {
     private static final String MOSCOW = "Moscow";
+    private static final String MOSCOW_LOWER_CASE = "moscow";
```
Comment (gregorusanov) [inline src/test/java/com/ghilly/web/handler/CityHandlerTest.java]: bad dicion, get MOSCOW and transform it to lower case

File: src/test/java/com/ghilly/integrationTest/CountryDAORestControllerIntegrationSuccessfulTest.java [hunk 4]
```diff
@@ -103,9 +103,9 @@ public void getCountryStatusOk() throws Exception {
 
     @Test
     public void updateCountryStatusOk() throws Exception {
-        CountryDAO countryDAO = new CountryDAO("USSR");
+        CountryDAO countryDAO = new CountryDAO("ussr");
         countryRepository.save(countryDAO);
-        int id = Objects.requireNonNull(countryRepository.findByName("USSR").orElse(null)).getId();
+        int id = Objects.requireNonNull(countryRepository.findByName(countryDAO.getName()).orElse(null)).getId();
```
Comment (gregorusanov) [inline src/test/java/com/ghilly/integrationTest/CountryDAORestControllerIntegrationSuccessfulTest.java]: revert it

File: src/test/java/com/ghilly/web/handler/CountryHandlerTest.java [hunk 5]
```diff
@@ -17,7 +17,9 @@
 class CountryHandlerTest {
     private static final int COUNTRY_ID = 1;
     private static final String NAME = "Russia";
+    private static final String NAME_LOWER_CASE = "russia";
     private static final CountryDAO RUS = new CountryDAO(COUNTRY_ID, NAME);
+    private static final CountryDAO RUS_FOR_SERVICE = new CountryDAO(COUNTRY_ID, NAME_LOWER_CASE);
```
Comment (gregorusanov) [inline src/test/java/com/ghilly/web/handler/CountryHandlerTest.java]: please rename it

**Review by gregorusanov — CHANGES_REQUESTED** [review]

File: src/main/java/com/ghilly/web/handler/CountryHandler.java (hunk 1 above)
Comment (ghilyana) [reply src/main/java/com/ghilly/web/handler/CountryHandler.java]: done

File: src/main/java/com/ghilly/web/handler/CountryHandler.java (hunk 2 above)
Comment (ghilyana) [reply src/main/java/com/ghilly/web/handler/CountryHandler.java]: done

File: src/test/java/com/ghilly/web/handler/CityHandlerTest.java (hunk 3 above)
Comment (ghilyana) [reply src/test/java/com/ghilly/web/handler/CityHandlerTest.java]: done

File: src/test/java/com/ghilly/web/handler/CountryHandlerTest.java (hunk 5 above)
Comment (ghilyana) [reply src/test/java/com/ghilly/web/handler/CountryHandlerTest.java]: done

File: src/test/java/com/ghilly/integrationTest/CountryDAORestControllerIntegrationSuccessfulTest.java (hunk 4 above)
Comment (ghilyana) [reply src/test/java/com/ghilly/integrationTest/CountryDAORestControllerIntegrationSuccessfulTest.java]: got it

**Review by gregorusanov — APPROVED** [review]

**Pull request merged**